src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -332,17 +357,30 @@ void Run (DirectoryAssemblyResolver res)
 			string applicationTemplateFile = "ApplicationRegistration.java";
 			SaveResource (applicationTemplateFile, applicationTemplateFile, real_app_dir,
 				template => template.Replace ("// REGISTER_APPLICATION_AND_INSTRUMENTATION_CLASSES_HERE", regCallsWriter.ToString ()));
+
+#if ENABLE_MARSHAL_METHODS
+			void StoreMarshalAssemblyPath (string name, ITaskItem asm)
+			{
+				if (Debug) {
+					return;
+				}
+
+				if (!marshalMethodsAssemblyPaths.TryGetValue (name, out HashSet<string> assemblyPaths)) {
+					assemblyPaths = new HashSet<string> ();
+					marshalMethodsAssemblyPaths.Add (name, assemblyPaths);
+				}
+
+				assemblyPaths.Add (asm.ItemSpec);
+			}
+#endif
 		}
 
-		bool CreateJavaSources (IEnumerable<TypeDefinition> javaTypes, TypeDefinitionCache cache)
+		bool CreateJavaSources (IEnumerable<TypeDefinition> javaTypes, TypeDefinitionCache cache, DirectoryAssemblyResolver resolver, MarshalMethodsClassifier classifier)
```

Comment:
The `DirectoryAssemblyResolver` doesn't appear to be used here, is it needed?

Reply:
Good catch, it's not anymore (was in an earlier incarnation of the code)